core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -178,7 +177,7 @@ protected[core] abstract class EntitlementService(config: WhiskConfig, loadBalan
         }
 
         entitlementCheck andThen {
-            case Success(r) =>
+            case Success(r) if resources.nonEmpty =>
```

Comment:
it's `andThen` simply for logging; it skips the success case where resource is empty.